webapp/src/main/java/com/box/l10n/mojito/service/ai/translation/AITranslateJob.java

```diff
@@ -0,0 +1,257 @@
+                      translateLocales(input, tmTextUnit, repository);
+                    } else {
+                      logger.debug(
+                          "Text unit with name: {} should not be translated, skipping AI translation.",
+                          tmTextUnit.getName());
+                      meterRegistry.counter(
+                          "AITranslateJob.translate.notTranslatable",
+                          Tags.of("repository", repository.getName()));
+                    }
+                  });
+          meterRegistry
+              .timer("AITranslateJob.timeToMT", Tags.of("repository", repository.getName()))
+              .record(Duration.between(JSR310Migration.dateTimeNow(), pendingMT.getCreatedDate()));
+        } else {
+          // If the pending MT is expired, log an error and delete it
+          logger.error("Pending MT for tmTextUnitId: {} is expired", input.getTmTextUnitId());
+          meterRegistry.counter(
+              "AITranslateJob.expired", Tags.of("repository", repository.getName()));
+        }
+      } else {
+        logger.warn("No pending MT found for tmTextUnitId: {}", input.getTmTextUnitId());
+        meterRegistry.counter(
+            "AITranslateJob.noPendingMTFound", Tags.of("repository", repository.getName()));
+      }
+    } catch (Exception e) {
+      logger.error("Error running job for text unit id {}", input.getTmTextUnitId(), e);
+      meterRegistry.counter(
+          "AITranslateJob.error", Tags.of("repository", input.getRepositoryId().toString()));
+    } finally {
+      TmTextUnitPendingMT pendingMT =
+          tmTextUnitPendingMTRepository.findByTmTextUnitId(input.getTmTextUnitId());
+      if (pendingMT != null) {
+        logger.debug("Deleting pending MT for tmTextUnitId: {}", input.getTmTextUnitId());
+        tmTextUnitPendingMTRepository.delete(pendingMT);
+      }
+    }
+    return null;
+  }
+
+  private void translateLocales(
+      AITranslateJobInput input, TMTextUnit tmTextUnit, Repository repository) {
+
+    Map<String, RepositoryLocaleAIPrompt> repositoryLocaleAIPrompts =
+        repositoryLocaleAIPromptRepository
+            .getActiveTranslationPromptsByRepository(input.getRepositoryId())
+            .stream()
+            .collect(
+                Collectors.toMap(
+                    rlap ->
+                        rlap.getLocale() != null
+                            ? rlap.getLocale().getBcp47Tag()
+                            : REPOSITORY_DEFAULT_PROMPT,
+                    Function.identity()));
+    repository.getRepositoryLocales().stream()
+        .filter(rl -> rl.getParentLocale() != null)
+        .map(RepositoryLocale::getLocale)
+        .forEach(
+            targetLocale -> {
+              try {
+                if (isExistingVariant(tmTextUnit, targetLocale)) {
+                  // If a translation already exists, skip the AI translation (likely exists due to
```

Comment:
Interesting point!

Should we ensure upstream that leveraged translation explicitly skip the AI Translation step somehow? (As in, they never get this far in the first place / they get filtered out earlier?)

Reply:
I think it's an improvement we could add on in future, particularly if we decide to paralellize the text unit AI translations per locale . 

Right now the leveraging steps don't report back what was leveraged, so I wanted to avoid going into that unrelated flow and introducing changes. Plus by doing the check in the AITranslateJob we centralize the AI translation logic, the job is set per text unit and then handles everything around getting to get that particular text unit AI translated.